cmd/deploy/deploy.go

```diff
@@ -197,13 +201,45 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, opts *Option
 		log.Infof("could not create temporal kubeconfig %s", err)
 		return err
 	}
-
 	var err error
-	// Read manifest file with the commands to be executed
-	opts.Manifest, err = dc.getManifest(cwd, contextCMD.ManifestOptions{Name: opts.Name, Filename: opts.ManifestPath})
-	if err != nil {
-		log.Infof("could not find manifest file to be executed: %s", err)
-		return err
+	if opts.Build && contextCMD.IsManifestV2Enabled() {
+		opts.Manifest, err = contextCMD.GetManifestV2(cwd, opts.ManifestPath)
+		if err != nil {
+			return err
+		}
+		log.Debug("found okteto manifest")
+
+		if opts.Manifest.Deploy == nil {
+			return fmt.Errorf("found okteto manifest, but no deploy commands where defined")
+		}
+
+		if opts.Manifest.Build != nil {
+			log.Debug("running build from manifest definition")
+
+			var buildErrs []string
+
+			for service, buildInfo := range opts.Manifest.Build {
+				buildOptions := build.BuildOptions{}
+				opts := build.OptsFromManifest(service, buildInfo, buildOptions)
+
+				ctx := context.Background()
```

Comment:
just curiosity, why do we need to create a new context for each build? I think you can use the one received on `runDeploy`

Reply:
i was using the same code from the loop at the build command, i will remove it.